src/vm/method.cpp

```diff
@@ -5055,6 +5055,37 @@ LPVOID NDirectMethodDesc::FindEntryPoint(HINSTANCE hMod) const
 }
 #endif // CROSSGEN_COMPILE
 
+#if !defined(CROSSGEN_COMPILE)
+//*******************************************************************************
+void NDirectMethodDesc::InitEarlyBoundNDirectTarget()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        INJECT_FAULT(COMPlusThrowOM(););
+    }
+        CONTRACTL_END
+
+        _ASSERTE(IsEarlyBound());
+
+    if (IsClassConstructorTriggeredAtLinkTime())
+    {
+        GetMethodTable()->CheckRunClassInitThrowing();
+    }
+
+    const void *target = GetModule()->GetInternalPInvokeTarget(GetRVA());
+    _ASSERTE(target != 0);
+
+    if (HeuristicDoesThisLookLikeAGetLastErrorCall((LPBYTE)target))
+        target = (BYTE*)FalseGetLastError;
```

Comment:
This is kind of weird, the pointer to the target can also hold the error state? Which is then set as the target.